kafka-handler/src/test/org/apache/hadoop/hive/kafka/AvroBytesConverterTest.java

```diff
@@ -0,0 +1,155 @@
+
+    Assert.assertNotNull(simpleRecordDeserialized);
+    Assert.assertEquals(simpleRecord, simpleRecordDeserialized);
+  }
+
+  /**
+   * Tests the default case of no skipped bytes per record works properly. 
+   */
+  @Test
+  public void convertWithAvroBytesConverter() {
+    // Since the serialized version was created by Confluent, lets remove the first five bytes to get the actual message.
+    byte[] simpleRecordWithNoOffset = Arrays.copyOfRange(simpleRecordConfluentBytes, 5, simpleRecordConfluentBytes.length);
+
+    Schema schema = SimpleRecord.getClassSchema();
+    KafkaSerDe.AvroBytesConverter conv = new KafkaSerDe.AvroBytesConverter(schema);
+    runConversionTest(conv, simpleRecordWithNoOffset);
+  }
+
+  /**
+   * Tests that the skip converter skips 5 bytes properly, which matches what Confluent needs.
+   */
+  @Test
+  public void convertWithConfluentAvroBytesConverter() {
+    Schema schema = SimpleRecord.getClassSchema();
+    KafkaSerDe.AvroSkipBytesConverter conv = new KafkaSerDe.AvroSkipBytesConverter(schema, 5);
+    runConversionTest(conv, simpleRecordConfluentBytes);
+  }
+
+  /**
+   * Tests that the skip converter skips a custom number of bytes properly.
+   */
+  @Test
+  public void convertWithCustomAvroSkipBytesConverter() {
+    int offset = 2;
+    // Remove all but two bytes of the five byte offset which Confluent adds, 
+    // to simulate a message with only 2 bytes in front of each message.
+    byte[] simpleRecordAsOffsetBytes = Arrays.copyOfRange(simpleRecordConfluentBytes, 5 - offset, simpleRecordConfluentBytes.length);
+
+    Schema schema = SimpleRecord.getClassSchema();
+    KafkaSerDe.AvroSkipBytesConverter conv = new KafkaSerDe.AvroSkipBytesConverter(schema, offset);
+    runConversionTest(conv, simpleRecordAsOffsetBytes);    
+  }
+
+  /**
+   * Test that when we skip more bytes than are in the message, we throw an exception properly.
+   */
+  @Test
+  public void skipBytesLargerThanMessageSizeConverter() {
+    // The simple record we are serializing is two strings, that combine to be 7 characters or 14 bytes.
+    // Adding in the 5 byte offset, we get 19 bytes. To make sure we go bigger than that, we are setting
+    // the offset to ten times that value. 
+    int offset = 190;
+
+    Schema schema = SimpleRecord.getClassSchema();
+    KafkaSerDe.AvroSkipBytesConverter conv = new KafkaSerDe.AvroSkipBytesConverter(schema, offset);
+
+    exception.expect(RuntimeException.class);
+    exception.expectMessage("org.apache.hadoop.hive.serde2.SerDeException: " + 
+      "Skip bytes value is larger than the message length.");
+    runConversionTest(conv, simpleRecordConfluentBytes);    
+  }
+
+  /**
+  * Test that we properly parse the converter type, no matter the casing.
+  */
+  @Test
+  public void bytesConverterTypeParseTest() {
+    Map<String, KafkaSerDe.BytesConverterType> testCases = new HashMap<String, KafkaSerDe.BytesConverterType>() {{
```

Comment:
Nit: double brace.

This could be implemented as a parameterized test

Reply:
The double brace is actually the way this has to be written prior to Java 9: https://stackoverflow.com/questions/6802483/how-to-directly-initialize-a-hashmap-in-a-literal-way. Note that thread talks about some memory issues, but since this is being used for a small map in a test and not in production, it seems fine.

Also, you're correct, we could use a parameterized test. I don't see any reason to use such a heavy mechanism for such a simple test, however. There's no need to do multiple tests per set of parameters, nor are there more than two parameters necessary.